bxsolana/provider/ws.ts

```diff
@@ -62,233 +62,305 @@ import {
+    getOrderStatusStream = async (request: GetOrderStatusStreamRequest): Promise<AsyncGenerator<GetOrderStatusStreamResponse>> => {
+        const subscriptionId = await this.wsConnection.subscribe("GetOrderStatusStream", request)
+
+        this.manageGetStreamMaps("GetOrderStatusStream", subscriptionId)
+
+        return this.wsConnection.subscribeToNotifications(subscriptionId)
     }
 
-    //POST requests
-    postOrder(request: PostOrderRequest): Promise<PostOrderResponse> {
-        return this.wsSocketCall("PostOrder", request)
+    getRecentBlockHashStream = async (request: GetRecentBlockHashRequest): Promise<AsyncGenerator<GetRecentBlockHashResponse>> => {
+        const subscriptionId = await this.wsConnection.subscribe("GetRecentBlockHashStream", request)
+
+        this.manageGetStreamMaps("GetRecentBlockHashStream", subscriptionId)
+
+        return this.wsConnection.subscribeToNotifications(subscriptionId)
     }
 
-    postSubmit(request: PostSubmitRequest): Promise<PostSubmitResponse> {
-        return this.wsSocketCall("PostSubmit", request)
+    getQuotesStream = async (request: GetQuotesStreamRequest): Promise<AsyncGenerator<GetQuotesStreamResponse>> => {
+        const subscriptionId = await this.wsConnection.subscribe("GetQuotesStream", request)
+
+        this.manageGetStreamMaps("GetQuotesStream", subscriptionId)
+
+        return this.wsConnection.subscribeToNotifications(subscriptionId)
     }
 
-    postSubmitBatch(request: PostSubmitBatchRequest): Promise<PostSubmitBatchResponse> {
-        return this.wsSocketCall("PostSubmitBatch", request)
+    getPoolReservesStream = async (request: GetPoolReservesStreamRequest): Promise<AsyncGenerator<GetPoolReservesStreamResponse>> => {
+        const subscriptionId = await this.wsConnection.subscribe("GetPoolReservesStream", request)
+
+        this.manageGetStreamMaps("GetPoolReservesStream", subscriptionId)
+
+        return this.wsConnection.subscribeToNotifications(subscriptionId)
     }
 
-    postCancelOrder(request: PostCancelOrderRequest): Promise<PostCancelOrderResponse> {
-        return this.wsSocketCall("PostCancelOrder", request)
+    // Since the implementation of the subscription functions rely on the count incrementing on every subsequent call
+    // to the same stream, the cancel functionality operates on the assumption that the "streamNumber" corresponds to
+    // the specific call of the stream by the user
+
+    // example
+    // getOrderbooksStream (count === 1)
+    // getOrderbooksStream (count === 2)
+
+    // can cancel 1, or 2. If a nonvalid cancellation number is sent as an input, the promise will be rejected with a
+    // false boolean
+
+    cancelStreamByCount = async (streamName: string, streamNumber: number): Promise<boolean> => {
+        const countToSubscriptionID = this.streamToCountMap.get(streamName)
+
+        if (countToSubscriptionID) {
+            this.countToSubscriptionID.forEach((value: string, key: number) => {
```

Comment:
this struct never gets written to. perhaps you want to do `countToSubscriptionID.forEach(...)`?

Reply:
if so, you should delete this property

Reply:
@aspin `countToSubscriptionID` is the map used by `streamToCountMap`. it gets written to in manageGetStreamsMap. Am I missing something here?